include/jemalloc/internal/jemalloc_internal_inlines_c.h

```diff
@@ -337,4 +343,217 @@ imalloc_fastpath(size_t size, void *(fallback_alloc)(size_t)) {
 	return fallback_alloc(size);
 }
 
+JEMALLOC_ALWAYS_INLINE bool
+nonfast_aligned(const void *ptr) {
```

Comment:
I just realized that we already have `free_fastpath_nonfast_aligned`. In that case this isn't a good naming choice. Since we are checking for different alignments in `free_fastpath_nonfast_aligned`, we should still call out which part this alignment is for. Let's still call it `prof_sample_aligned`.

We definitely don't want to duplicate the code though; even if it's only 1 line, it's part of the core logic (and would cause super confusing bugs the two copies are out of sync). Can we remove the one in `prof_inlines.h` and only keep this one (maybe even include this header in prof_inlines.h)?

Reply:
I think we can. The function is used in both prof_inlines.h and jemalloc.c and they both can include this header. It seems my last comment asking this was not sent out. I wanted to ask about the duplicate. I agree that we should not duplicate, but if this is for prof purposes and start with a prefix `prof_`, will it be a bit weird to appear in this file and this file only?

Reply:
It's indeed not ideal. Maybe taking a step back and look into how the current header dependency is set up, and see if it's possible to keep it prof_inlines.h?

Even if we can't, it won't be too bad though, given that we also have other inline utilities included (e.g. the `tcache_` ones). Maintaining the DAG does require some hassles.

(I'm also open to still adding a new `_internal_inline` header too, if you feel that's the best / cleanest solution).

Reply:
I have tested multiple things and still go ahead with keeping `prof_sample_aligned` in `jemalloc_internal_inlines_c.h`. Although it's not ideal, I think it's still acceptable and relatively easy to trace. Besides, the "interdependency" I used before is not accurate. I think the problem here is that the include of different inline headers is strictly ordered [here](https://github.com/jemalloc/jemalloc/blob/dev/include/jemalloc/internal/jemalloc_internal_includes.h#L81). Inline headers included later, i.e., prof_inlines.h, are safe to include other inline headers listed before them, i.e., jemalloc_internal_inlines_c.h. Changing the order to keep `prof_sample_aligned` where it was is not worthy. I think the current modification is the cleanest I can find.